picard/ui/infodialog.py

```diff
@@ -290,6 +290,48 @@ def format_file_info(file_):
                              htmlescape(i[1])), info))
 
 
+def format_tracklist(cluster):
+    info = []
+    info.append("<b>%s</b> %s" % (_('Album:'),
+                                  htmlescape(cluster.metadata["album"])))
+    info.append("<b>%s</b> %s" % (_('Artist:'),
+                                  htmlescape(cluster.metadata["albumartist"])))
+    info.append("")
+    TrackListItem = namedtuple('TrackListItem', 'tracknumber, title, artist, length')
+    tracklists = defaultdict(list)
+    if isinstance(cluster, Album):
+        objlist = cluster.tracks
+    else:
+        objlist = cluster.iterfiles(False)
+    for obj_ in objlist:
+        m = obj_.metadata
+        artist = m["artist"] or m["albumartist"] or cluster.metadata["albumartist"]
+        track = TrackListItem(m["tracknumber"], m["title"], artist,
+                              m["~length"])
+        tracklists[obj_.discnumber].append(track)
+
+    def sorttracknum(item):
+        try:
+            return int(item.tracknumber)
```

Comment:
The int conversion is not needed here, the `tracknumber` property always returns an int. Hence we also never go into the exception here, tracknumber will be 0 instead. But this also means it fails in the "3/10" case. Do we have a bug here? I see the following options:

1. Keep the conversion as is, but use `m["tracknumber"]` directly here
2. Integrate the regex parsing into `Item.tracknumber` property (is this wanted at other places?)
3. Use `item.tracknumber`, but compare it to 0 and if it is do the regex parsing
4. Drop parsing the string

I tend to prefer 2, but depends on how else this property gets used.

Reply:
I think you missed something:
- `item.tracknumber` in this context is from metadata
- `item` is a TrackListItem named tuple in https://github.com/metabrainz/picard/blob/8491e3f92b078bc7a40f109ddc263da9c3ed0b00/picard/ui/infodialog.py#L378
- `discnumber` comes from `File` or `Track` `discnumber` property though

Reply:
Got it, thanks :) From my side I think this is good to merge, definitely makes the UI more consistent